### PR TITLE
:star: Add ZFS resources to OS provider

### PR DESCRIPTION
**Pull request by tas50**
## Summary
- Add `zfs`, `zfs.pool`, and `zfs.dataset` resources to the OS provider
- Pools expose health, sizeBytes, capacity, dedup ratio, and auto-expand/replace/trim settings
- Datasets cover filesystems, volumes, snapshots, and bookmarks with compression, quotas, encryption, and mount info
- Both `zfs.pool` and `zfs.dataset` support `init(name)` for targeted lookup and lazy `properties()` for the full property set
- `zfs.dataset.snapshots()` filters child snapshots from the cached dataset list
- All byte-valued fields use a `Bytes` suffix for clarity (e.g., `sizeBytes`, `usedBytes`, `quotaBytes`)

## Example queries
```coffee
zfs.pools { name health sizeBytes freeBytes }
zfs.datasets.where(type == "filesystem") { name usedBytes mountpoint }
zfs.pool(name: "rpool").properties
zfs.dataset(name: "rpool/data").snapshots { name creation }
```

## Test plan
- [x] Parser unit tests (13 tests covering pool list, dataset list, properties, edge cases)
- [x] Interactive verification on a ZFS system: `mql run local -c "zfs.pools { name health sizeBytes }"`
- [x] Verify `zfs.pool(name: "...")` init lookup works
- [x] Verify `zfs.dataset(name: "...").snapshots` filters correctly

### CODE REVIEW

File: providers/os/resources/zfs.go [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+		return "", err
+	}
+	cmd := o.(*mqlCommand)
+	if exit := cmd.GetExitcode(); exit.Data != 0 {
+		return "", errors.New("could not retrieve zfs version: " + cmd.Stderr.Data)
+	}
+	version := strings.TrimSpace(cmd.Stdout.Data)
+	if i := strings.IndexByte(version, '\n'); i != -1 {
+		version = version[:i]
+	}
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/os/resources/zfs.go:41]: Accepting this limitation

File: .github/actions/spelling/expect.txt [hunk 2]
```diff
@@ -2,17 +2,20 @@ aad
 ACCOUNTADMIN
 ACTIVEMQ
 advancedthreatprotection
+artifactregistry
```
[the remark this answers is not in the document]
Comment (tas50) [reply .github/actions/spelling/expect.txt:5]: These were moved to the main dictionary

[the remark this answers is not in the document]
Comment (tas50) [reply .github/actions/spelling/expect.txt:5]: that's intentional
